### ISSUE TITLE
Quick Start Guides - Incorrect Error Message

### ISSUE THREAD

**Issue by Alonski** [microsoft/TypeScript#15302]
**TypeScript Version:**  2.2.1 / nightly (2.2.0-dev.201xxxxx)

**Code**
In section *Type annotations* you are shown that trying to use [0, 1, 2] as a string gives an error.
While running this myself I got different behavior from the TS transpiler/compiler.
```ts
// A *self-contained* demonstration of the problem follows...
function greeter(person: string) {
    return "Hello, " + person;
}

var user = [0, 1, 2];

document.body.innerHTML = greeter(user);
```

**Expected behavior:**
```
greeter.ts(7,35): error TS2345: Argument of type 'number[]' is not assignable to parameter of type 'string'
```
**Actual behavior:**
```
greeter.ts(7,26): Supplied parameters do not match any signature of call target
```

**Comment by mhegazy** [comment]
A PR to the handbook (https://github.com/Microsoft/TypeScript-Handbook) would be appreciated.

**Comment by Alonski** [comment]
I will create a pull request soon. I read on the readme that I shouldn't make pull requests but instead open issues 😀

**Comment by DanielRosenwasser** [comment]
Where did you see that? In [`doc/README.md`](https://github.com/Microsoft/TypeScript/blob/a1a2006074a01ca0b75b08294377cc52302e3e71/doc/README.md), we say to only open issues on the spec. We redirect handbook users to the handbook repository.

**Comment by Alonski** [comment]
Ok maybe that's why I was mistaken thanks

**Comment by gcnew** [comment]
The old error message is more helpful and descriptive than the new one. Is the best resolution to fix the documentation? Looks like a regression to me.

**Comment by Alonski** [comment]
@gcnew But the new error message is the current error message that is thrown when using this code.

**Comment by gcnew** [comment]
@Alonski Yes, I understand that. I'm trying to start a discussion, because helpful messages are one of the major selling points of TypeScript. This is the simplest of function invokations - no overloads, no generics, no unions. My understandig is that the new error message is a serious degradation and maybe the problem should be solved in the compoler by bringing the old behaviour back.

**Comment by DanielRosenwasser** [comment]
@gcnew has a good point - the content was written that way to make it easier for users to understand what's going on. The real problem is that the compiler is gives bad error messages.

**Comment by RyanCavanaugh** [comment]
This works better now
